Reinforcement-learning use cases (#388)
huggingface/hub-docs: tasks/src/reinforcement-learning/about.md

> ## Use cases

Reviewer: You can directly copy and paste this template https://github.com/huggingface/hub-docs/blob/main/tasks/src/placeholder/about.md and make your changes inside instead 🙂

In this part we will see where we can actually use these algorithms and how it is helping the people in numerous ways.

### Applications in autonomous driving

Autonomous or self driving cars are quite trending now a days. It is also a big step towards driver less commute or where the machine is itself able to perform the task, which makes it an problem of machine learning and artificial intelligence. So the vehicle, driving in the real world without any driver must be backed by an machine learning model trained with all the possibilities. 

> In self-driving cars, there are various aspects to consider, such as speed limits at various places, drivable zones, avoiding collisions — just to mention a few. The collection of these variety of situations is tough problem to solve.

Reviewer: ```suggestion
In self-driving cars, there are various problems to consider, such as speed limits or drivable zones. The collection of these variety of situations is tough problem to solve.

```
grammar nit 🙂

> Reinforcement learning can be used to prepare a model for this purpose. Reinforcement Learning models are trained in a dynamic environment by learning a policy from its own experiences following the principles of exploration and exploitation that minimize disruption to traffic. Self-driving cars may learn from many aspects which will make it able to make a optimal decision when stuck in a situation.In this case we can also call it as scenario based learning.

Reviewer: ```suggestion
Reinforcement learning can be used to train a model for self-driving cars. Reinforcement Learning models are trained in a dynamic environment by learning a policy from its own experiences, following the principles of exploration and exploitation that minimize disruption to traffic. Self-driving cars can learn from many aspects which will make the car capable of making an optimal decision given a state. In this case we can also call it as scenario based learning.

```

> Reinforcement Learning models are trained in a dynamic environment by learning a policy from its own experiences, following the principles of exploration and exploitation that minimize disruption to traffic
Can you make this part into pieces to ease the readability and explain a bit more about how exploration/exploitation trade-off works in this case to minimize disruption? 

> Self-driving cars can learn from many aspects which will make it able to make a optimal decision when stuck in a situation.In this case we can also call it as scenario based learning.
Aspects of what exactly? 
Can you give a bit of an explanation on scenario based learning too?

> Some of the autonomous driving tasks where reinforcement learning could be applied include trajectory optimization, motion planning, dynamic pathing, controller optimization, and scenario-based learning policies for highways. Reinforcement Learning agents are trained in dynamic environments to optimize trajectories. The agents learn motion planning, route changing, decision and position of parking and speed control, etc.

Reviewer: ```suggestion
Some of the autonomous driving tasks where reinforcement learning could be applied include trajectory optimization, motion planning, dynamic path finding, controller optimization, and scenario-based learning policies for highways. Reinforcement Learning agents are trained in dynamic environments to optimize trajectories. The agents are capable of accomplishing tasks such as motion planning, route changing, decision and position of parking and speed control.

```

### NLP (Natural Language processing)

> First of all, NLP is that branch of machine learning or artificial intelligence which is concerned with giving computers the ability to understand the text and spoken words the same way as a human being does.  

Reviewer: ```suggestion
Natural Language Processing is the branch of computer science concerned with giving computers the ability to understand the text and spoken words.  

```
grammar nit 🙂

Language understanding uses Reinforcement Learning because of its inherent nature of decision making it learns from the mistakes. The agent tries to understand the state of the sentence and tries to form an action set maximizing the value it would add.

> Reinforcement Learning is used in multiple areas of NLP like text summarization, question answering, translation, dialogue generation, machine translation etc.  

Reviewer: ```suggestion
Reinforcement Learning is used in various NLP tasks such as text summarization, question answering, translation, dialogue generation, or machine translation.

```
Can you elaborate more on this?

> Reinforcement Learning agents can be trained to understand a few sentences of the document and use it to answer the corresponding questions. Reinforcement Learning with a combination of RNN is used to generate the answers for those questions as shown in this paper.

Reviewer: Instead of linking to a paper, can you simply put how they can do that? Our audience is non-ML experts who want to build products without dealing with the technicalities such as architecture.

> ### Applications in Healthcare 

Reviewer: We try to be careful about life critical use cases, just saying 🙂

Reinforcement learning can be used to guide patients to the right treatment regime or by choosing the optimized plan for the treatment.

> Reinforcement Learning in healthcare is categorized as dynamic treatment regimes(DTRs) in chronic disease or critical care, automated medical diagnosis, and other general domains. The medical sector has always used technological advancements, for past few years it has used the reinforcement learning for implementing dynamic treatment regimes(DTRs) for patients who have long term illness i.e. chronic disease.

Reviewer: ```suggestion
Reinforcement Learning in healthcare is used in dynamic treatment regimes in chronic disease or critical care and automated medical diagnosis.

```
This kinda seems like a life critical use case 😅

> It has also found its application in automated medical diagnosis, health resource scheduling, drug discovery and development, and health management.Reinforcement Learning has taken over medical report generation, identification of nodules/tumors and blood vessel blockage, analysis of these reports, etc.

Reviewer: ```suggestion
Reinforcement learning is also used in health resource scheduling, drug discovery and development, and health management. More specifically, reinforcement learning is also used in medical report generation, identification of nodules/tumors and blood vessel blockage, analysis of these reports.
```
Can you elaborate how models can do that?

## Task Description

## Most used dataset for the task

## Most used model for the task

## Metrics that are used to evaluate the task

## Libraries used for the task

## Use cases

> In this part we will see where we can actually use these algorithms and how it is helping the people in numerous ways.

Reviewer: ```suggestion
In this part, we will see where we can use these algorithms and how it is helping people in numerous ways.

```

### Applications in autonomous driving

> Autonomous or self driving cars are quite trending nowadays. It is also a big step towards driverless commute where the machine is itself able to perform the task. The vehicle without any driver can be backed by a machine learning model. 

Reviewer: ```suggestion
Autonomous or self-driving cars are pretty trending nowadays. It is also a big step towards driverless commute where the machine is itself able to perform the task. A machine learning model can back the vehicle without any driver. 

```

> In self-driving cars, there are various problems to consider, such as speed limits or drivable zones. The collection of these variety of situations is tough problem to solve.

Reviewer: ```suggestion
In self-driving cars, there are various problems, such as speed limits or drivable zones. The collection of this variety of situations is a challenging problem to solve.

```

Reinforcement learning can be used to train a model for self-driving cars. Reinforcement Learning models are trained in a dynamic environment by learning a policy from its own experiences, following the principles of exploration and exploitation that minimize disruption to traffic. 

The exploration-exploitation trade-off is a fundamental dilemma whenever you learn about the world by trying things out. The dilemma is between choosing what you know and getting something close to what you expect (‘exploitation’) and choosing something you aren’t sure about and possibly learning more (‘exploration’). 

Exploitation consists of taking the decision assumed to be optimal with respect to the data observed so far. This SAFE approach tries to avoid bad decisions as much as possible but also prevents from discovering potential better decisions.

Exploration consists of not taking the decision that seems to be optimal, assuming that the available data is not sufficient to find the truly optimal solution. This more RISKY approach can sometimes lead to poor decisions but also makes it possible to discover better ones, if there exists any.

> Deep Reinforcement learning can learn by itself to operate autonomously in extreme conditions. The dynamic double deep Q-learning (DDQN) model enables the proposed system not be confined to only in known environments. The exploration and exploitation strategies of DDQN enables the autonomous agent to learn proper decisions for various dynamic environments and tracks. The proposed model is tested in a gaming environment. It shows the overall effectiveness in traversing of autonomous land vehicles. The goal is to make the system effective to traverse through undiscovered parts by detecting obstacles.

Reviewer: This part is plagiarised from this link it seems: https://ieeexplore.ieee.org/document/9016539 
Can you please avoid plagiarism? (and also provide credits when you refer to a document?) We don't want a fancy language in these documents anyway, we want to include beginner audience 🙂

> Scenario based methods for testing and validation of automated driving systems (ADS) in virtual test environments are gaining importance and becoming an essential component for verification and validation processes of ADS. The increase in the cost and complexity of real testing lead to an highly increased efforts in real world testing. Using scenario and simulation based approaches this effort can be efficiently reduced with respect to costs and time. Research has shown that it is necessary to drive and test billions of kilometers to ensure safety of ADS which would not be possible considering the time and cost effort for real testing. The biggest challenges are the selection of a suitable simulation framework and the selection of relevant scenarios for the system under test.

Reviewer: Same with this link: https://www.researchgate.net/publication/344896382_Scenario_Based_Testing_of_Automated_Driving_Systems_A_Literature_Survey

### Robotics 

Robots operate in a highly dynamic and ever-changing environment, making it impossible to predict what will happen next. Reinforcement Learning provides a considerable advantage in these scenarios to make the robots robust enough and help acquire complex behaviors adaptively in different scenarios.

It aims to remove the need for time-consuming and tedious checks and replaces them with computer vision systems ensuring higher levels of quality control on the production assembly line.

A wide variety of problems in robotics can be considered as ones of reinforcement learning. Reinforcement learning (RL) enables a robot to autonomously discover an optimal behavior through trial-and-error interactions with its environment. Instead of explicitly detailing the solution to a problem, in reinforcement learning the designer of a control task provides feedback in terms of a scalar objective function that measures the one-step performance of the robot.

## Task Variants 

### Q - Learning

* It is used to learn the policy for reinforcement learning
* Policy: a rule that the agent should follow to select actions given the current state.

> * Q - learning is to find the optimal policy for making decisions

Reviewer: isn't Q-learning a value based technique rather than policy based? Also pinging @simoninithomas

Reviewer 2: Q-Learning is valued based but since you find the best value function and the policy is argmax q value you're "indirectly" find an optimal policy function.

## Task Description

this place describes the task. Contribute by adding a nice description!

## Most used dataset for the task

Write about the data used to train a reinforcement learning model. As reinforcement learning model create its own data, explain about that. 

## Most used model for the task

Contribute by giving example of the model that is used for this task.

## Metrics that are used to evaluate the task

Explain different evaluation techniques which are used to find the underlying weakness of reinforcement learning models.

> ## Libraries used for the task

Reviewer: We don't have a section for this.

## Glossary

> 

Reviewer: ```suggestion
```

> ## Use cases

Reviewer: ```suggestion
## Use Cases

```
it's a header 🙂

### NLP (Natural Language processing)

NLP is a branch of machine learning or artificial intelligence concerned with allowing computers to understand the text and spoken words the same way as a human does.  

Language understanding uses Reinforcement Learning because of its inherent nature of decision making it learns from the mistakes. The agent tries to understand the state of the sentence and tries to form an action set maximizing the value it would add.

Reinforcement Learning is used in various NLP tasks such as text summarization, question answering, translation, dialogue generation, or machine translation.

Reinforcement Learning agents can be trained to understand a few sentences of the document and use it to answer the corresponding questions.

The state space S is a set of all the states that the agent can transition to and action space A is a set of all actions the agent can act out in a certain environment. There are also Partial Observable cases, where the agent is unable to observe the complete state information of the environment.

> #### Q - Learning

Reviewer: I think this section can be task variant.

* it will find the best course of action, given the current state of the agent. Depending on where the agent is in the environment, it will decide the next action to be taken.

With the help of Q - value and advanced statistics we can find the Deep - Q - Network. You all must have heard of a game called chess. nowadays it is mostly played online, also you don't need a second player to play, you can just play against a bot and also get defeated, which leaves you thinking how can a machine beat me. Well, because it is learning. 

Now, the thing to understand is when training the learner for games like Chess, the agent has a very small action space, but large state space. For example, in the game of chess, the state space is the whole 20X20 board, but the action space is small like a pawn can move only in 1–2 directions. Deep-Q-Network has been proven to be very effective.

But when we talk about reinforcement learning for natural language processing there are few problems which are discussed below:

#### Sequential decision making problem for text understanding

> * Agent observes state as a string of text at a time t

Reviewer: Would be nice to have it as a paragraph

A wide variety of problems in robotics can be considered problems related to reinforcement learning. Reinforcement learning enables a robot to autonomously discover an optimal behavior by interacting with its environment.

## Task Variants 

### Q - Learning

Q - learning will find the best course of action, given the agent's current state. Depending on where the agent is in the environment, it will decide the following action to be taken. It chooses the action such that it maximizes the reward.

#### Sequential decision making problem for text understanding

Agent observes state as a string of text at a time 't'. Agent also knows a set of possible actions, each describes as a string text. Agent tries to understand the “state text” and all possible “action texts”, then takes a decision which maximizes the long term reward. Then, the environment state changes to a new state, agent receives an immediate reward.

#### Unbounded action space in reinforcement learning

> Environments with vast state space is a problem in reinforcement learning. Also the action space can be quite large. 

Reviewer: ```suggestion
Environments with vast state space are a problem in reinforcement learning. Also, the action space can be pretty large. 

```
Readability nit :)

> For solving this problem DRRN(deep reinforcement relevance network) was proposed. The idea of DRRN is to project both the state and action into a continuous space(as vectors). Q-function is a relevance function of the state vector and action vector.the DRRN experiences higher average reward.

Reviewer: ```suggestion
To solve this problem, DRRN (deep reinforcement relevance network) was proposed. The idea of DRRN is to project both the state and action into a continuous space (as vectors). Q-function is a relevance function of the state vector and action vector. The DRRN experiences a higher average reward.

```
If we are going to have this can you please explain more how it works?

Reviewer 2: It's the first time I hear about relevance network. I was wondering if it's something "too advanced for the task page? 🤔

**Reward:** The numerical feedback signal that the agent receives from the environment based on the chosen action

> 

Reviewer: ```suggestion
```

**Return:** Cumulative Reward.  In the simplest case, the return is the sum of the rewards.

> 

Reviewer: ```suggestion
```

**Episode:** For some applications there is a natural notion of final time step.  In this case, there is a starting point and an ending point (a terminal state). This creates an episode: a list of States, Actions, Rewards, and new States. For instance, think about Chess: an episode begins at the initial board position and ends when the game is over.

> 

Reviewer: ```suggestion
```

**Policy:** The Policy is the brain of the Agent, it’s the function that tells what action to take given the state. So it defines the agent’s behavior at a given time. Reinforcement learning methods specify how the agent’s policy is changed as a result of its experience.

## Use Cases

> 

Reviewer: ```suggestion
```

### Applications in autonomous driving

> 

Reviewer: ```suggestion
```

The exploration-exploitation trade-off is a fundamental dilemma whenever you learn about the world by trying things out. The dilemma is between choosing what you know and getting something close to what you expect (‘exploitation’) and choosing something you aren’t sure about and possibly learning more (‘exploration’). 

The deep reinforcement learning algorithms can learn to operate in unknown conditions. The dynamic-double deep Q-learning model makes the system to not be limited only in known environments. The exploration-exploitation trade-off enables the system to make decisions in lesser known environments. 

Some of the autonomous driving tasks where reinforcement learning could be applied include trajectory optimization, motion planning, dynamic path finding, controller optimization, and scenario-based learning policies for highways. Reinforcement Learning agents are trained in dynamic environments to optimize trajectories. The agents are capable of accomplishing tasks such as motion planning, route changing, decision and position of parking and speed control.

> ### NLP (Natural Language processing)

Reviewer: ```suggestion
### Natural Language Processing

```

> 

Reviewer: ```suggestion
```

Robots in industries or healthcare working towards reducing manual intervention use reinforcement learning to map natural language instructions to sequences of executable actions. During training, the learner repeatedly constructs action sequences, executes those actions, and observes the resulting rewards. A reward function works in the backend that defines the quality of these executed actions.

### Robotics 

Robots operate in a highly dynamic and ever-changing environment, making it impossible to predict what will happen next. Reinforcement Learning provides a considerable advantage in these scenarios to make the robots robust enough and help acquire complex behaviors adaptively in different scenarios. It aims to remove the need for time-consuming and tedious checks and replaces them with computer vision systems ensuring higher levels of quality control on the production assembly line.

A wide variety of problems in robotics can be considered problems related to reinforcement learning. Reinforcement learning enables a robot to autonomously discover an optimal behavior by interacting with its environment.

## Task Variants 

> ### Q - Learning

Reviewer: ```suggestion
### Q-Learning

```

> Q - learning will find the best course of action, given the agent's current state. Depending on where the agent is in the environment, it will decide the following action to be taken. It chooses the action such that it maximizes the reward.

Reviewer: ```suggestion
Q-learning will find the best course of action, given the agent's current state. Depending on where the agent is in the environment, it will decide the following action to be taken. It chooses the action such that it maximizes the reward.

```

The exploration-exploitation trade-off is a fundamental dilemma whenever you learn about the world by trying things out. The dilemma is between choosing what you know and getting something close to what you expect (‘exploitation’) and choosing something you aren’t sure about and possibly learning more (‘exploration’). 

The deep reinforcement learning algorithms can learn to operate in unknown conditions. The dynamic-double deep Q-learning model makes the system to not be limited only in known environments. The exploration-exploitation trade-off enables the system to make decisions in lesser known environments. 

Some of the autonomous driving tasks where reinforcement learning could be applied include trajectory optimization, motion planning, dynamic path finding, controller optimization, and scenario-based learning policies for highways. Reinforcement Learning agents are trained in dynamic environments to optimize trajectories. The agents are capable of accomplishing tasks such as motion planning, route changing, decision and position of parking and speed control.

### NLP (Natural Language processing)

> NLP is a branch of machine learning or artificial intelligence concerned with allowing computers to understand the text and spoken words the same way as a human does.  

Reviewer: ```suggestion
Natural language processing is a branch of machine learning concerned with allowing computers to understand text and spoken words the same way a human does.  

```

> Language understanding uses Reinforcement Learning because of its inherent nature of decision making it learns from the mistakes. The agent tries to understand the state of the sentence and tries to form an action set maximizing the value it would add.

Reviewer: ```suggestion
Dialogue systems use reinforcement learning because of its inherent nature of decision-making. It learns from mistakes. The system tries to understand the sentence's state and form an action set maximizing the value it would add.

```
it's not only the understanding that uses RL, it's usually that it's dialogue management system that chooses actions not intent recognition part

Reinforcement Learning is used in various NLP tasks such as text summarization, question answering, translation, dialogue generation, or machine translation.

Reinforcement Learning agents can be trained to understand a few sentences of the document and use it to answer the corresponding questions.

> Robots in industries or healthcare working towards reducing manual intervention use reinforcement learning to map natural language instructions to sequences of executable actions. During training, the learner repeatedly constructs action sequences, executes those actions, and observes the resulting rewards. A reward function works in the backend that defines the quality of these executed actions.

Reviewer: Can I have the source for this?

### Robotics 

Robots operate in a highly dynamic and ever-changing environment, making it impossible to predict what will happen next. Reinforcement Learning provides a considerable advantage in these scenarios to make the robots robust enough and help acquire complex behaviors adaptively in different scenarios.

It aims to remove the need for time-consuming and tedious checks and replaces them with computer vision systems ensuring higher levels of quality control on the production assembly line.

> A wide variety of problems in robotics can be considered as ones of reinforcement learning. Reinforcement learning (RL) enables a robot to autonomously discover an optimal behavior through trial-and-error interactions with its environment. Instead of explicitly detailing the solution to a problem, in reinforcement learning the designer of a control task provides feedback in terms of a scalar objective function that measures the one-step performance of the robot.

Reviewer: ```suggestion
A wide variety of problems in robotics can be considered as problems related to reinforcement learning. Reinforcement learning enables a robot to autonomously discover an optimal behavior, through interactions with its environment. Instead of explicitly detailing the solution to a problem, the designer of a control task provides feedback in terms of a scalar objective function that measures the one-step performance of the robot.

```
Can you simplify the last sentence for a broader audience?

## Task Variants 

### Q - Learning

* It is used to learn the policy for reinforcement learning
* Q-function[Q(s,a)]: returns Q-value for action a at state s. Learning an action-value function, a.k.a. Q function, that computes the expected utility of taking an action in a state after training converges.
* it will find the best course of action, given the current state of the agent. Depending on where the agent is in the environment, it will decide the next action to be taken.

With the help of Q - value and advanced statistics we can find the Deep - Q - Network. You all must have heard of a game called chess. nowadays it is mostly played online, also you don't need a second player to play, you can just play against a bot and also get defeated, which leaves you thinking how can a machine beat me. Well, because it is learning. 

> Now, the thing to understand is when training the learner for games like Chess, the agent has a very small action space, but large state space. For example, in the game of chess, the state space is the whole 20X20 board, but the action space is small like a pawn can move only in 1–2 directions. Deep-Q-Network has been proven to be very effective.

Reviewer: Deep Q learning is very ineffective for games where branching factor is very high. You need to use MCTS. Deep Q learning is good for environments that has continuous variables (e.g. cartpole) so I'd rather not include chess as an example. (You can check out the solutions of DeepMind for Go)

But when we talk about reinforcement learning for natural language processing there are few problems which are discussed below:

#### Sequential decision making problem for text understanding

Agent observes state as a string of text at a time 't'. Agent also knows a set of possible actions, each describes as a string text. Agent tries to understand the “state text” and all possible “action texts”, then takes a decision which maximizes the long term reward. Then, the environment state changes to a new state, agent receives an immediate reward.

#### Unbounded action space in reinforcement learning

> Not only the state space is huge, but the action space is also huge too. Action is characterized by unbounded natural language descriptions. For example, if say to the model “Hi! how are you? Have you eaten your lunch? Don't forget to call me tomorrow as it is the bigday ”. Well, this input text from me is the state-space for the model(quite heavy), and the action space is every text combination available(or infinity). This problem for such a huge action space was still the problem in Deep-Q-Network.

Reviewer: ```suggestion
Environments with vast state space is a problem in reinforcement learning. On top of this, action space can be enormous too, and choosing responses to natural language prompts is one of them. 

```
Can you rephrase the rest of your writing like I did above? It was a bit too unofficial.

> For solving this problem DRRN(deep reinforcement relevance network) was proposed. The idea of DRRN is to project both the state and action into a continuous space(as vectors). Q-function is a relevance function of the state vector and action vector.the DRRN experiences higher average reward.

Reviewer: I see it's taken from here, again, we really don't need academic content. We want beginners to understand too! https://paperswithcode.com/paper/deep-reinforcement-learning-with-a-natural/review/

### Robotics 

Robots operate in a highly dynamic and ever-changing environment, making it impossible to predict what will happen next. Reinforcement Learning provides a considerable advantage in these scenarios to make the robots robust enough and help acquire complex behaviors adaptively in different scenarios.

It aims to remove the need for time-consuming and tedious checks and replaces them with computer vision systems ensuring higher levels of quality control on the production assembly line.

A wide variety of problems in robotics can be considered as ones of reinforcement learning. Reinforcement learning (RL) enables a robot to autonomously discover an optimal behavior through trial-and-error interactions with its environment. Instead of explicitly detailing the solution to a problem, in reinforcement learning the designer of a control task provides feedback in terms of a scalar objective function that measures the one-step performance of the robot.

## Task Variants 

### Q - Learning

* It is used to learn the policy for reinforcement learning

> * Policy: a rule that the agent should follow to select actions given the current state.

Reviewer: This is already defined in the glossary.

> The deep Reinforecement Learning finds application in a chatbot dialogue. Conversations are simulated using two virtual agents and the quality is improved in progressive iterations.

Reviewer: ```suggestion
Chatbot dialogues can be improved using Deep Reinforcement Learning. Conversations are simulated using two virtual agents, and the quality is improved in progressive iterations.

```
Can you please give a reference to this? I don't think this is used.

> ## Glossary

Reviewer: I think it would be good to put this section above Use Cases given it's essential to understand use cases.

Reviewer: Can you address this comment?

**State:** Information the agent receives from the environment. In the case of a video game it can be a frame (a screenshot), in the case of a chess playing agent it can be the board position, in the case of a trading agent it can be the price of a certain stock.

**Action:** The decision taken by the agent.

**Reward:** The numerical feedback signal that the agent receives from the environment based on the chosen action

**Return:** Cumulative Reward.  In the simplest case, the return is the sum of the rewards.

**Episode:** For some applications there is a natural notion of final time step.  In this case, there is a starting point and an ending point (a terminal state). This creates an episode: a list of States, Actions, Rewards, and new States. For instance, think about Chess: an episode begins at the initial board position and ends when the game is over.

**Policy:** The Policy is the brain of the Agent, it’s the function that tells what action to take given the state. So it defines the agent’s behavior at a given time. Reinforcement learning methods specify how the agent’s policy is changed as a result of its experience.

> ## Inference

Reviewer: Can you put the explanation in placeholder page under this?

Reviewer: Can you address this comment?

- [Train a Deep Reinforcement Learning lander agent to land correctly on the Moon 🌕 using Stable-Baselines3](https://github.com/huggingface/deep-rl-class/blob/main/unit1/unit1.ipynb)
- [Introduction to Unity MLAgents](https://colab.research.google.com/github/huggingface/deep-rl-class/blob/main/unit4/unit4.ipynb)
- [Training Decision Transformers with 🤗 transformers](https://github.com/huggingface/blog/blob/main/notebooks/101_train-decision-transformers.ipynb)

> This page was made possible thanks to the efforts of [Ram Ananth](https://huggingface.co/RamAnanth1), [Emilio Lehoucq](https://huggingface.co/emiliol) and [Osman Alenbey](https://huggingface.co/osman93).

Reviewer: Why did you remove this?

- [Train a Deep Reinforcement Learning lander agent to land correctly on the Moon 🌕 using Stable-Baselines3](https://github.com/huggingface/deep-rl-class/blob/main/unit1/unit1.ipynb)
- [Introduction to Unity MLAgents](https://colab.research.google.com/github/huggingface/deep-rl-class/blob/main/unit4/unit4.ipynb)
- [Training Decision Transformers with 🤗 transformers](https://github.com/huggingface/blog/blob/main/notebooks/101_train-decision-transformers.ipynb)

> 

Reviewer: Can you remove the additional spaces?
